BDArmory/Modules/BDModulePilotAI.cs

```diff
@@ -2304,27 +2304,31 @@ void UpdateTerrainAlertDetectionRadius(Vessel v)
                 belowMinAltitude = true; // Inform other parts of the code to behave as if we're below minimum altitude.
 
                 float maxAngle = 70.0f * Mathf.Deg2Rad; // Maximum angle (towards surface normal) to aim.
-                if (BDArmorySettings.SPACE_HACKS)
-                {
-                    maxAngle = 180.0f * Mathf.Deg2Rad;
-                }
                 float adjustmentFactor = 1f; // Mathf.Clamp(1.0f - Mathf.Pow(terrainAlertDistance / terrainAlertThreatRange, 2.0f), 0.0f, 1.0f); // Don't yank too hard as it kills our speed too much. (This doesn't seem necessary.)
                                              // First, aim up to maxAngle towards the surface normal.
-                Vector3 correctionDirection = Vector3.RotateTowards(terrainAlertDirection, terrainAlertNormal, maxAngle * adjustmentFactor, 0.0f);
-                // Then, adjust the vertical pitch for our speed (to try to avoid stalling).
-                if (!BDArmorySettings.SPACE_HACKS) //no need to worry about stalling in null atmo
+                if (BDArmorySettings.SPACE_HACKS) //no need to worry about stalling in null atmo
+                {
+                    FlyToPosition(s, vessel.transform.position + upDirection * 100); //so point nose stright up for maximum vertical thrust. Future SI - test out using terrainAlertNormal instead
```

Comment:
Yeah, you should use `terrainAlertNormal`, otherwise you'll drift sideways right into vertical walls.